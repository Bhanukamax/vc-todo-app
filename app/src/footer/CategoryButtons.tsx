import React from "react";
import styled from "styled-components";
import { Link } from "@reach/router";

export const CategoryButtons: React.FC = () => (
  <Container>
    <p>View: </p>
    <StyledLink to="/">All</StyledLink>
    <StyledLink to="/active">Active</StyledLink>
    <StyledLink to="/completed">Completed</StyledLink>
  </Container>
);

const NavLink = (props: any) => (
  <Link
    {...props}
    getProps={({ isCurrent }) => {
      return {
        style: {
          color: isCurrent ? "#fff" : "#666",
          backgroundColor: isCurrent ? "#83daca" : "#eaeaea"
        }
      };
    }}
  />
);

const Container = styled.div`
  display: flex;
`;

const StyledLink = styled(NavLink)`
  border: 0;
  padding: 10px 20px;
  margin: 0 10px;
  border-radius: 25px;
  font-size: 1em;
  text-decoration: none;
  line-height: 2em;
  display: inline-block;
  :hover {
    background: #83daca;
    color: #fff;
    cursor: pointer;
  }
`;
