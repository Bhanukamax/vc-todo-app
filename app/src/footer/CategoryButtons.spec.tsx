import React from "react";
import "jest-styled-components";
import { CategoryButtons, NavLink, StyledLink } from "./CategoryButtons";
import { shallow } from "enzyme";

describe("<CategoryButtons/>", () => {
  it("renders correctly", () => {
    const component = shallow(<CategoryButtons />);

    expect(component).toMatchSnapshot();
  });

  it("renders the children propertly", () => {
    const Wrapper = shallow(<CategoryButtons />);

    expect(Wrapper.find(StyledLink)).toHaveLength(3);
  });
});
