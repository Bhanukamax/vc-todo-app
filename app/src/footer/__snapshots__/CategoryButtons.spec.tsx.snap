// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryButtons/> renders correctly 1`] = `
<styled.div>
  <p>
    View: 
  </p>
  <Styled(NavLink)
    to="/"
  >
    All
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/active"
  >
    Active
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/completed"
  >
    Completed
  </Styled(NavLink)>
</styled.div>
`;
